### PR TITLE
Aarch64 MMU implementation

### PR DESCRIPTION
**Pull request by Deryugin**
Now it supports 4KiB, 16KiB and 64KiB page granules.

Also this PR fixes some memory issues (i.e. hard-coded 4KiB page size or missing headers)

### CODE REVIEW

File: src/arch/aarch64/include/asm/hal/env/traps_core.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/**
+ * @file traps_core.h
```
Comment (anton-bondarev) [inline src/arch/aarch64/include/asm/hal/env/traps_core.h]: is this file used anywhere?

File: src/arch/aarch64/include/asm/hal/reg.h [hunk 2]
```diff
@@ -170,6 +179,18 @@ static inline uint64_t aarch64_ttbr0_el2_read(void) {
...
+static inline uint64_t aarch64_ttbr0_el1_read(void) {
+	return read_system_reg(TTBR0_EL1);
+}
+
+static inline void aarch64_ttbr0_el1_write(uint64_t reg) {
+	write_system_reg(TTBR0_EL1, reg);
+}
+
 static inline uint64_t aarch64_ttbr1_el2_read(void) {
 	return 0;//read_system_reg(TTBR1_EL2);
```
Comment (anton-bondarev) [inline src/arch/aarch64/include/asm/hal/reg.h]: read_system_reg(TTBR1_EL2); brakes the system?

File: src/drivers/interrupt/cortex_a9_gic/cortex_a9_gic.c [hunk 3]
```diff
@@ -175,4 +175,5 @@ void swi_handle(void) {
 	printk("swi!\n");
 }
 
-PERIPH_MEMORY_DEFINE(gic, GIC_CPU_BASE, 0x2020);
+PERIPH_MEMORY_DEFINE(gic_cpu, GIC_CPU_BASE, 0x2020);
+PERIPH_MEMORY_DEFINE(gic_distributor, GIC_DISTRIBUTOR_BASE, 0x2020);
```
Comment (anton-bondarev) [inline src/drivers/interrupt/cortex_a9_gic/cortex_a9_gic.c]: Size is also 0x2020?

Comment (Deryugin) [reply src/drivers/interrupt/cortex_a9_gic/cortex_a9_gic.c]: Fixed.

File: src/arch/aarch64/include/asm/hal/env/traps_core.h (hunk 1 above)
Comment (Deryugin) [reply src/arch/aarch64/include/asm/hal/env/traps_core.h]: It's required for `mmu_core` test.

File: src/arch/aarch64/include/asm/hal/reg.h (hunk 2 above)
Comment (Deryugin) [reply src/arch/aarch64/include/asm/hal/reg.h]: Fixed.

**Pull request merged**